src/analysis/convex/strict_convex_space.lean

```diff
@@ -238,3 +238,149 @@ lemma norm_midpoint_lt_iff (h : ∥x∥ = ∥y∥) : ∥(1/2 : ℝ) • (x + y)
+  have hn : ∥f (-x)∥ = ∥-(f x)∥,
+  { rw [hi.norm_map_of_map_zero h0, norm_neg, norm_neg, hi.norm_map_of_map_zero h0] },
+  rw [←same_ray_iff_of_norm_eq hn, same_ray_iff_norm_add, ←sub_eq_add_neg, ←dist_eq_norm,
+      hi.dist_eq, dist_eq_norm, hi.norm_map_of_map_zero h0, norm_neg, norm_neg,
+      hi.norm_map_of_map_zero h0, sub_eq_add_neg, ←neg_add, norm_neg],
+  exact (same_ray.refl x).norm_add
+end
+
+lemma map_smul_nonneg_of_map_zero {f : F → E} (hi : isometry f) (h0 : f 0 = 0) {r : ℝ}
+  (hr : 0 ≤ r) (x : F) :
+  f (r • x) = r • f x :=
+begin
+  have hn : ∥f (r • x)∥ = ∥r • f x∥,
+  { rw [hi.norm_map_of_map_zero h0, norm_smul, norm_smul,
+        hi.norm_map_of_map_zero h0] },
+  rw ←same_ray_iff_of_norm_eq hn,
+  refine same_ray.nonneg_smul_right _ hr,
+  rw [same_ray_iff_norm_add, ←neg_neg (f x), ←sub_eq_add_neg, ←hi.map_neg_of_map_zero h0,
+      ←dist_eq_norm, hi.dist_eq, dist_eq_norm, sub_neg_eq_add, norm_neg,
+      hi.norm_map_of_map_zero h0, hi.norm_map_of_map_zero h0, norm_neg],
+  exact (same_ray_nonneg_smul_left x hr).norm_add
+end
+
+lemma map_smul_of_map_zero {f : F → E} (hi : isometry f) (h0 : f 0 = 0) (r : ℝ) (x : F) :
+  f (r • x) = r • f x :=
+begin
+  rcases le_or_lt 0 r with (h|h),
+  { exact hi.map_smul_nonneg_of_map_zero h0 h x },
+  { rw [←neg_neg r, neg_smul, hi.map_neg_of_map_zero h0, neg_smul (-r)],
+    congr,
+    rw ←neg_pos at h,
+    exact hi.map_smul_nonneg_of_map_zero h0 h.le x }
+end
+
+lemma map_add_eq_smul_sub_map_sub {f : F → E} (hi : isometry f) (x y : F) :
+  f (x + y) = (2 : ℕ) • f x - f (x - y) :=
+begin
+  set g : F → E := λ v, f (x + v) - f x with hg,
+  have hg0 : g 0 = 0, { rw hg, simp },
+  have hfg : ∀ v, f (x + v) = g v + f x, { simp [hg] },
+  have hig : isometry g,
+  { intros u v, simp [hg, hi.edist_eq] },
+  rw [sub_eq_add_neg x, hfg, hfg, hig.map_neg_of_map_zero hg0],
+  abel
+end
+
+lemma map_add_eq_smul_sub_map_sub_rev {f : F → E} (hi : isometry f) (x y : F) :
+  f (x + y) = (2 : ℕ) • f y - f (y - x) :=
+by rw [add_comm, hi.map_add_eq_smul_sub_map_sub]
+
+lemma map_add_of_map_zero {f : F → E} (hi : isometry f) (h0 : f 0 = 0) (x y : F):
+  f (x + y) = f x + f y :=
+calc f (x + y) = (2⁻¹ : ℝ) • (2 : ℝ) • f (x + y) : by simp
+     ...       = (2⁻¹ : ℝ) • (f (x + y) + f (x + y)) : by simp [two_smul]
+     ...       = (2⁻¹ : ℝ) • ((2 : ℕ) • f x - f (x - y) + ((2 : ℕ) • f y - f (y - x))) :
+      by rw [←hi.map_add_eq_smul_sub_map_sub, ←hi.map_add_eq_smul_sub_map_sub_rev]
+     ...       = f x + f y :
+      begin
+        rw [←neg_sub x, hi.map_neg_of_map_zero h0, sub_neg_eq_add, sub_add_add_cancel, smul_add,
+            two_smul, two_smul, ←two_smul ℝ (f x), ←two_smul ℝ (f y), ←mul_smul, ←mul_smul],
+        simp
+      end
+
+/-- An isometry of real normed spaces with strictly convex codomain is a linear isometry if it
+maps 0 to 0.  Unlike Mazur-Ulam, this does not require the isometry to be surjective.  -/
+def linear_isometry_of_map_zero {f : F → E} (hi : isometry f) (h0 : f 0 = 0) :
```

Comment:
Given that `isometry` is an unbundled predicate, this will probably be more convenient.
```suggestion
def linear_isometry_of_map_zero (f : F → E) (hi : isometry f) (h0 : f 0 = 0) :
```

Reply:
Done, and thus removed the `show` that was previously needed to provide a type when this definition was used.